walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,159 @@
 
 package com.google.sps;
 
+import com.google.common.collect.Sets;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  /*
+   * Takes in a list of non-overlapping time ranges sorted by start (and end) as well as a new
+   * event. Returns a list of non-overlapping time ranges covered by the existing list as well
+   * as the given event.
+   */
+  private ArrayList<TimeRange> addToBusy(ArrayList<TimeRange> busyTimes, Event meeting) {
+    TimeRange meetingTime = meeting.getWhen();
+
+    /*
+     * Stores the index of the search key, if it is contained in the list; otherwise,
+     * (-(insertion point) - 1). The insertion point is defined as the point at which
+     * the key would be inserted into the list: the index of the first element greater
+     * than the key, or list.size() if all elements in the list are less than the specified key.
+     * This variable signifies where the start time of the meeting fits into the list.
+     */
+    int startIndex = Collections.binarySearch(busyTimes, meetingTime, TimeRange.ORDER_BY_START);
+    // The position where the end time of the meeting fits into the sorted list
+    int endIndex = Collections.binarySearch(busyTimes, meetingTime, TimeRange.ORDER_BY_END);
+    // Finding the actual position from the above value, which may be negative
+    int startPos = (startIndex < 0) ? (-(startIndex + 1)) : startIndex;
+    int prev = startPos - 1;
+    // Finding the nonnegative position of the next element
+    int next = (endIndex < 0) ? (-(endIndex + 1)) : endIndex;
+    ArrayList<TimeRange> newBusyTimes = new ArrayList<>();
+
+    // stores the index in the array from which this meeting begins
+    int startPoint = 0;
```

Comment:
`startPoint` and `startTime` feels like the same attribute to me. Maybe `startIndex` instead of `startPoint`?